src/Reflection/ReflectionParameter.php

```diff
@@ -497,8 +496,26 @@ private function getClassName(): ?string
         }
 
         $type = $this->getType();
-        assert($type instanceof ReflectionType);
-        $typeHint = (string) $type;
+        assert($type instanceof ReflectionNamedType || $type instanceof ReflectionUnionType);
+
+        if ($type instanceof ReflectionUnionType) {
```

Comment:
Wait, WTF is happening here? :D 

In a union, `getClassName` shouldn't make sense 🤔

Reply:
Fun with ~flags~ PHP: https://3v4l.org/jQ1oJ